### DISCUSSION TITLE
Generate an icosphere in libigl

### DISCUSSION THREAD

**Discussion by lukkio88** [libigl/libigl#2184] [Q&A]
Hello,

Is there a way to construct an icosphere in libigl at different resultions?
I know there's an algorithm I could implement from scratch but before I do that is there maybe a functionality of libigl that does that for me?

Thank you

**Comment by alecjacobson** [comment]
This function in gptoolbox is super useful https://github.com/alecjacobson/gptoolbox/blob/master/mesh/subdivided_sphere.m
https://github.com/alecjacobson/gptoolbox/blob/master/external/icosahedron.m

We should include a port to libigl. Basically igl::icosohedron and then igl::subdivided_sphere that calls that and igl::loop

Happy to merge a PR if you feel motivated to translate those.

**Comment by lukkio88** [comment]
I ended up implementing my own, I can commit that if you want. Although I see more options in your version.

**Comment by alecjacobson** [comment]
Sure that'd be great
